runtime/src/comm/gasnet/comm-gasnet.c

```diff
@@ -759,17 +759,20 @@ static void set_max_segsize() {
     return;
   }
 
-  // If GASNET_NEEDS_MAX_SEGSIZE is defined then we have to have
-  // GASNET_MAX_SEGSIZE set.  Otherwise, we don't.
-#ifdef GASNET_NEEDS_MAX_SEGSIZE
+  // If segment is large or fast, we need to set GASNET_MAX_SEGSIZE.
+  // Note: this used to be ifdef GASNET_NEEDS_MAX_SEGSIZE
```

Comment:
It's unusual to leave a comment behind to describe how the code used to work -- is there a particular reason for doing it this time given git's (or whatever's) ability to show history?

Reply:
I've moved those changes to another branch (per @ronawho's request) and I'll leave him in charge of carrying them forward in whatever form makes sense.